centralstore/redis_store.go

```diff
@@ -113,10 +125,13 @@ func (r *RedisBasicStore) WriteSpan(span *CentralSpan) error {
 	case DecisionDrop:
 		return nil
 	case DecisionKeep, AwaitingDecision:
-		_ = r.traces.incrementSpanCounts(conn, span.TraceID, span.Type)
+		err := r.traces.incrementSpanCounts(conn, span.TraceID, span.Type)
+		if err != nil {
+			r.errs <- err
```

Comment:
If r.errs is full because no one is reading from it, this will block forever. Perhaps you want to write to this chan as a select with a default that drops the error?